### PR TITLE
Added User schema and refered in context and customer attribute in fu…

### PR DESCRIPTION
**Pull request by venkatramanm**
…lfillment. There would not be a backwards compatibility issue with this addition. user attribute may be passed in context to make intelligent search responses based on user's location

### CODE REVIEW

**Comment by beckn-dev** [comment]
User schema cannot be transmitted in headers as it violates the principle of optimal ignorance. You can pass it in the message body as an encrypted token.

**Comment by venkatramanm** [comment]
According to the discussion in today's call, We will keep it open till a larger community discusses and comes up with the solution to passing user information and optimizing the gw to send to a small set of bpps.

**Comment by beckn-dev** [comment]
Reopened this thread to discuss further. @venkatramanm, given that beckn is a transaction protocol, can you please elaborate on what occasion the details of the user of a BAP needs to be transmitted to the BPP? On a beckn-enabled network, the purpose of a BPP is to acquire _orders_ and not _users_.

**Comment by beckn-dev** [comment]
Secondly, when you mentioned, " user attribute may be passed in context to make intelligent search responses based on user's location", the location is already being passed in the `Fulfillment` object as Fulfillment.start.location.gps or Fulfillment.end.location.gps. If a BPP wants to return search results basis the user's location, he can use the above values.

File: core/v0/api/core.yaml [hunk 1]
```diff
@@ -1943,6 +1940,17 @@ components:
         time:
           $ref: '#/components/schemas/Time'
 
+    User:
```
Comment (core-wg-admin) [inline core/v0/api/core.yaml:1943]: User information strictly remains with the BAP and must only be accessed by the BPP via a consent framework like DEPA. In face, the personal details like name, gender, contact info `Person` schema should also be removed from the Core spec and should only allow an ID with a consent token to be transmitted that authorizes the BPP to fetch specific details from the BAP.

**Comment by core-wg-admin** [comment]
BG optimization is more of a Registry implementation recomendation and does not necessarily form part of a transaction API. To allow fewer broadcast endpoints, a network may allow higher granulariy on discovery but that is not the purpose of the BG. The purpose of a BG is to allow equal opportunity for BPPs to respond to a search Intent.

**Comment by core-wg-admin** [comment]
Closing this PR for now as it requires further discussion and deliveration. Let us continue this in the discussion forum and then re-submit this PR.

**Pull request closed without merging**